File: drift-static-profiler/tests/fixtures/go-gin/handlers/orders_handler.go
package handlers

import (
	"encoding/json"
	"net/http"

	"example.com/orders/service"
)

type OrdersHandler struct {
	svc *service.OrdersService
}

func NewOrdersHandler(s *service.OrdersService) *OrdersHandler {
	return &OrdersHandler{svc: s}
}

type createOrderDTO struct {
	CustomerEmail string `json:"customerEmail"`
	TotalCents    int64  `json:"totalCents"`
	Currency      string `json:"currency"`
}

func (h *OrdersHandler) CreateOrder(w http.ResponseWriter, r *http.Request) {
	var dto createOrderDTO
	if err := json.NewDecoder(r.Body).Decode(&dto); err != nil {
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}
	order, err := h.svc.CreateOrder(dto.CustomerEmail, dto.TotalCents, dto.Currency)
	if err != nil {
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(order)
}

File: drift-static-profiler/tests/fixtures/kotlin-ktor/src/main/kotlin/com/example/handlers/OrdersHandler.kt
package com.example.handlers

import com.example.services.OrdersService

data class CreateOrderDto(
    val customerEmail: String,
    val totalCents: Long,
    val currency: String?,
)

class OrdersHandler(private val svc: OrdersService) {

    fun createOrder(dto: CreateOrderDto): Long {
        return svc.createOrder(dto.customerEmail, dto.totalCents, dto.currency)
    }
}
